### DISCUSSION TITLE
On interoperability with solid

### DISCUSSION THREAD

**Discussion by damooo** [ontola/atomic-server#670] [General]
As atomic data seems to be subset of rdf, can a specific profile of solid that allows restricted form be interoperable with apps or any of the components of the project?

I currently released [Manas solid server](https://github.com/manomayam/manas/), which almost implements solid storage spec, (notifications protocol is WIP). It  has oidc for resource server authn, WAC/ACP for hierarchichal authorization (reusable crates), n3 patch, etc. And soon will have pods management, indexing,  etc.

I am wondering any reuse/adaptation is possible in either way.

Thanks for your work.

**Comment by joepio** [comment]
Thanks for reaching out! 

In theory it's possible to build an ontology using Atomic Data Classes + Properties, and build something very similar to Solid from there. But it will not work with any existing Solid data, because the property URLs in existing RDF data are just too unpredictable. Some URLs don't even resolve, some resolve to an HTML page, some resolve to RDF documents, most contain no schema definitions (so they are completely type-unsafe). My [critiques on RDF](https://docs.atomicdata.dev/interoperability/rdf.html) + Solid have been the prime motivator for starting Atomic Data.

But again, it is possible _if you implement the classes and properties using Atomic Data Classes and Properties_. Currently my colleague @Polleps is working on an Ontology Editor #648 #655 which could make the process of converting Solid concepts into Atomic possible.

Note that I do expect some other barriers as well (how to edit resources, authentication, etc). Open to help solve these issues though!

**Comment by damooo** [comment]
It will be great to have all the great applications being developed can be adaptable to solid. I.e if their core part works on hierarchy of documents, and we can layer/plugin authn/authz etc.

There is really need for an ontology editor and other essential goodies. But it is too much work to redo them all. I should once explore how the apps works, and what can be done.

**Reply by joepio** [reply]
Sounds cool :) join the [discord](https://discord.gg/a72Rv2P) if you wanna do a video call sometime
